File: data_structures/python/Graph.py
class Vertex:
    def __init__(self, n):
        self.name = n
        self.neighbors = {}

    def add_neighbor(self, v, weight):
        if v not in self.neighbors:
            self.neighbors[v] = weight

class Graph:
    def __init__(self):
        self.vertices = {}

    def addVertex(self, v):
        if v not in self.vertices:
            self.vertices[v] = Vertex(v)

    def addEdge(self, u, v, weight=0):
        if u in self.vertices and v in self.vertices:
            self.vertices[u].add_neighbor(v, weight)
            self.vertices[v].add_neighbor(u, weight)

    def printGraph(self):
        for key, value in self.vertices.items():
            print('{}: {}'.format(key, value.neighbors))


g = Graph()
g.addVertex('A')
g.addVertex('B')
g.addVertex('C')
g.addEdge('A', 'B', 1)
g.addEdge('A', 'C', 2)
g.addEdge('C', 'B', 3)
g.printGraph()

File: data_structures/c++/Graph.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

class Vertex
{
public:
	string name;
	map<string, int> neighbors;

	Vertex() {};
	Vertex(string n) : name(n) {};

	void add_neighbors(string v, int weight)
	{
		neighbors.insert(pair<string, int>(v, weight));
	}

};


class Graph
{
public:
	map<string, Vertex> vertices;

	Graph() {};

	void add_vertex(string v)
	{
		vertices.insert(pair<string, Vertex>(v, Vertex(v)));
	}
	void add_edge(string u, string v, int weight)
	{
		if (vertices.find(u) != vertices.end() && vertices.find(v) != vertices.end()) {
			vertices[u].add_neighbors(v, weight);
			vertices[v].add_neighbors(u, weight);
		}
	}

	void printNeighbors(string v)
	{
		for (auto& t : vertices[v].neighbors)
			std::cout << t.first << " : " << t.second << ", ";
	}

	void printGraph()
	{

		for (auto& t : vertices) {
			std::cout << t.first << " = ";
			printNeighbors(t.first);
			std::cout << "\n";
		}
	}
};

int main()
{
	Graph g;
	g.add_vertex("A");
	g.add_vertex("B");
	g.add_vertex("C");
	g.add_edge("A", "B", 1);
	g.add_edge("A", "C", 2);
	g.add_edge("C", "B", 3);
	g.printGraph();
}